include/gutter_tree.h

```diff
@@ -57,11 +56,15 @@ class BufferTree {
    * @param level       the level of the buffer being flushed (0 is root)
    * @returns nothing
    */
-  flush_ret_t do_flush(char *data, uint32_t size, uint32_t begin, 
+  flush_ret_t do_flush(flush_struct &flush_from, uint32_t size, uint32_t begin, 
     node_id_t min_key, node_id_t max_key, uint16_t options, uint8_t level);
 
-  // Circular queue in which we place leaves that fill up
-  CircularQueue *cq;
+  // Work queue in which we place leaves that fill up
+  WorkQueue *wq;
+
+  flush_ret_t flush_control_block(BufferControlBlock *bcb);
+  flush_ret_t flush_internal_node(BufferControlBlock *bcb);
+  flush_ret_t flush_leaf_node(BufferControlBlock *bcb);
```

Comment:
my IDE complains these are unused and un-implemented

Reply:
They're definitely both implemented and used

Reply:
Can you point to where in the code that happens? All I see are the 2-argument versions

Reply:
I see the issue now. I'll fix it